packages/react/ds-app-launchpad/src/ui/MarkdownEditor/MarkdownEditor.tsx

```diff
@@ -0,0 +1,188 @@
+          />
+        )}
+        {!hideToolbar && (
+          <Toolbar label="Markdown Editor">
+            <ToolbarGroup label="Text Formatting">
+              <ToolbarButton
+                label={toolbar.title.label}
+                onClick={toolbar.title.handler}
+                shortcut={toolbar.title.shortcut}
+              >
+                {icons.ToolbarTitle}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.bold.label}
+                onClick={toolbar.bold.handler}
+                shortcut={toolbar.bold.shortcut}
+              >
+                {icons.ToolbarBold}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.italic.label}
+                onClick={toolbar.italic.handler}
+                shortcut={toolbar.italic.shortcut}
+              >
+                {icons.ToolbarItalic}
+              </ToolbarButton>
+            </ToolbarGroup>
+            <ToolbarSeparator />
+            <ToolbarGroup label="Tools">
+              <ToolbarButton
+                label={toolbar.quote.label}
+                onClick={toolbar.quote.handler}
+                shortcut={toolbar.quote.shortcut}
+              >
+                {icons.ToolbarQuote}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.code.label}
+                onClick={toolbar.code.handler}
+                shortcut={toolbar.code.shortcut}
+              >
+                {icons.ToolbarCode}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.link.label}
+                onClick={toolbar.link.handler}
+                shortcut={toolbar.link.shortcut}
+              >
+                {icons.ToolbarLink}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.uList.label}
+                onClick={toolbar.uList.handler}
+                shortcut={toolbar.uList.shortcut}
+              >
+                {icons.ToolbarUnorderedList}
+              </ToolbarButton>
+              <ToolbarButton
+                label={toolbar.oList.label}
+                onClick={toolbar.oList.handler}
+                shortcut={toolbar.oList.shortcut}
+              >
+                {icons.ToolbarOrderedList}
+              </ToolbarButton>
+            </ToolbarGroup>
+          </Toolbar>
+        )}
+      </div>
+
+      <textarea
+        defaultValue={defaultValue}
+        placeholder={placeholder}
+        className="editor-content"
+        ref={textareaRef}
+        style={style}
+        hidden={editMode !== "write"}
+      />
+      {editMode === "preview" && (
+        <div className="editor-content" style={style} ref={previewRef}>
+          <ReactMarkdown
+            rehypePlugins={[rehypeSanitize]}
+            components={{
+              h1: ({ children }) => <h3>{children}</h3>,
+              h2: ({ children }) => <h3>{children}</h3>,
```

Comment:
Is this for semantic page structure reasons, so that the markdown editor never has a heading level less than 3? This could benefit from a short comment explaining the rationale

Reply:
Not much reasoning there, except it would be too distractive visually if h1 and h2 where to be rendered here.